airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java

```diff
@@ -202,33 +196,37 @@ public void close() throws Exception {
 
   }
 
+  @Slf4j
   static class GlobalMemoryManager {
 
-    public static final long BLOCK_SIZE_BYTES = 10 * 1024 * 1024;
-    private long currentMemoryBytes = 0L;
+    private static final long BLOCK_SIZE_BYTES = 10 * 1024 * 1024; // 10MB
     private final long maxMemoryBytes;
 
-    // buffers
+    private final AtomicLong currentMemoryBytes = new AtomicLong(0);
 
     public GlobalMemoryManager(final long maxMemoryBytes) {
       this.maxMemoryBytes = maxMemoryBytes;
     }
 
     public synchronized long requestMemory() {
-      if (currentMemoryBytes >= maxMemoryBytes) {
+      if (currentMemoryBytes.get() >= maxMemoryBytes) {
         return 0L;
       }
 
-      final var freeMem = maxMemoryBytes - currentMemoryBytes;
+      final var freeMem = maxMemoryBytes - currentMemoryBytes.get();
       // Never allocate more than free memory size.
       final var toAllocateBytes = Math.min(freeMem, BLOCK_SIZE_BYTES);
-      currentMemoryBytes += toAllocateBytes;
+      currentMemoryBytes.addAndGet(toAllocateBytes);
 
       return toAllocateBytes;
     }
 
     public void free(final long bytes) {
```

Comment:
```suggestion
    public void synchronized free(final long bytes) {
```

Reply:
what calls 'free'?

Reply:
why does this need to be synchronized? i get it for request memory. but not for free.

Reply:
wait figured out what calls free

Reply:
I believe free is called in each worker pool thread

Reply:
yeah, but it is thread safe. it's just updating an atomiclong. there's no reason why 2 threads can't do that at the same time.

Reply:
you are right